### ISSUE TITLE
install failure due to 

### ISSUE THREAD

**Issue by erkinalp** [jehna/humanify#71]
```
npm WARN deprecated gauge@4.0.4: This package is no longer supported.
npm WARN deprecated are-we-there-yet@3.0.1: This package is no longer supported.
npm WARN deprecated npmlog@6.0.2: This package is no longer supported.
npm ERR! code 1
npm ERR! path /usr/local/lib/node_modules/humanifyjs/node_modules/isolated-vm
npm ERR! command failed
npm ERR! command sh -c prebuild-install || (node-gyp rebuild --release -j max && node-gyp clean)
```

**Comment by 0xdevalias** [comment]
See also:

- https://github.com/jehna/humanify/issues/52

---

I don't think those deprecated packages are actually the source of your install failure here though; I think it's more related to `isolated-vm`.

Are you able to share more details about what system you are running this on, what version of node, etc?

I know `isolated-vm` is used by @j4k0xb 's `webcrack` (which this project uses), and I know I have seen other `isolated-vm` issues there before, so it might also help to have a look on there too:

- https://github.com/j4k0xb/webcrack/issues?q=sort%3Aupdated-desc+isolated-vm

**Comment by erkinalp** [comment]
Node 18.19
Ubuntu 24.04 (x86)
GNOME 46

**Comment by jehna** [comment]
Yep, you'll need node.js >= 20, the prebuilt binaries are available for those versions only.

**Comment by jehna** [comment]
I've found thw [nodesource distributions](https://github.com/nodesource/distributions?tab=readme-ov-file#installation-instructions-deb) to be a straightforward way to install more recent node versions on debian

**Comment by jehna** [comment]
There is a note about needing node.js 20 in the readme, but it seems to be easy to miss. I wonder which place would be better. Meybe a preinstall hook thst ensures the version?

**Comment by 0xdevalias** [comment]
> I wonder which place would be better.

@jehna Does the `package.json` `engines` field enforce it?

- https://docs.npmjs.com/cli/v10/configuring-npm/package-json#engines
  - > engines
    > You can specify the version of node that your stuff works on
  - > Unless the user has set the engine-strict config flag, this field is advisory only and will only produce warnings when your package is installed as a dependency.
- https://stackoverflow.com/questions/28409883/how-to-specify-enforce-a-specific-node-js-version-to-use-in-package-json

Sounds like it may only work when installed as a dependency, so maybe not when installed directly?

**Comment by neoOpus** [comment]
> There is a note about needing node.js 20 in the readme, but it seems to be easy to miss. I wonder which place would be better. Maybe a preinstall hook thst ensures the version?

So, you advise using this in WSL instead of directly on Windows NodeJS ?

**Comment by jehna** [comment]
Unfortunately I have no idea how Windows works on development, the last time I used Windows as my main dev machine was over 10 years ago 😅 

I'd guess both WSL and non-wsl should work fine, as long as you have recent enough version. You can use e.g. [nvm](https://github.com/nvm-sh/nvm) to switch between versions on WSL
